internal/pkg/plugininstaller/kubectl/installer.go

```diff
@@ -0,0 +1,111 @@
+package kubectl
+
+import (
+	"os"
+	"time"
+
+	"github.com/cenkalti/backoff"
+
+	"github.com/devstream-io/devstream/internal/pkg/plugininstaller"
+	"github.com/devstream-io/devstream/internal/pkg/plugininstaller/util"
+	"github.com/devstream-io/devstream/pkg/util/kubectl"
+)
+
+// InstallByDownload will download file for apply
+func InstallByDownload(downloadUrl string) plugininstaller.BaseOperation {
+	processFunc := func(options plugininstaller.RawOptions) error {
+		// generate k8s config file for apply
+		configFileName, err := createKubectlFile(downloadUrl, "", options)
+		defer os.Remove(configFileName)
+		if err != nil {
+			return err
+		}
+		// kubectl apply -f
+		if err = kubectl.KubeApply(configFileName); err != nil {
+			return err
+		}
+		return nil
+	}
+	return processFunc
+}
+
+// InstallByContent will use input content for apply
+func InstallByContent(content string) plugininstaller.BaseOperation {
+	processFunc := func(options plugininstaller.RawOptions) error {
+		// generate k8s config file for apply
+		configFileName, err := createKubectlFile("", content, options)
+		if err != nil {
+			return err
+		}
+		defer os.Remove(configFileName)
```

Comment:
Defer after error return, if `createKubectlFile` encounter errors, the temp file won't be removed.